spec/integration/post_show_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :system do
+  describe 'displays detailed page of a single post' do
+    it 'displays a single post' do
+      # navigate to the posts page
+      visit user_posts_path(1)
+
+      # assert the displayed page
+      assert_text 'hardware'
+
+      sleep(1)
+      click_on 'hardware'
+
+      # page.click_link(:id => 'post_link')
```

Comment:
- [ ] to keep your repo clean kindly remove any commented code from your project 🙏